### PR TITLE
Use values in forms

### PR DESCRIPTION
**Pull request by kalashnikovisme**
- [x] Refactored form builder to use form builder methods. The reason for the refactoring. In a previous [PR](https://github.com/Purple-Magic/tramway/pull/37) I changed the implementation of TramwayForm Builder to use `ActionView::Base` form helpers instead of `FormBuilder` methods (both ActionView::Base and FormBuilder have their own `text_field`, `select`, .etc methods). So, I've faced some issues in forwarding the form options to ActionView::Base form helpers calls. So, I decided to send the object of class `Method` to the ViewComponent (`input` variable), so be able to call FormBuilder `text_field` directly from ViewComponent.
- [x] Fixed getting values to inputs from options or object

This refactoring was conducted to the beat of [this song](https://www.youtube.com/watch?v=IoBP24I2lwA). Please conduct your code review to the same tune, for the true harmony of programming! La la-laaa la la, la-la la-la la-laaaa!  ❤️❤️❤️❤️❤️❤️

### CODE REVIEW

File: app/components/tailwinds/form/builder.rb [hunk 1]
```diff
@@ -29,11 +41,18 @@ def submit(action, **options, &)
 
       private
 
+      def input(method_name)
+        unbound_method = self.class.superclass.instance_method(method_name)
```
Comment (moshinaan) [inline app/components/tailwinds/form/builder.rb:45]: why you need variable here?
```suggestion
        self.class.superclass.instance_method(method_name).bind(self)
```

File: spec/components/tailwinds/form/builder_spec.rb [hunk 2]
```diff
@@ -7,13 +7,32 @@
...
+      let(:value) { 'leopold@purple-magic.com' }
+      let(:output) { builder.text_field :email, value: }
+
+      it 'gets value from options' do
+        expect(output).to have_selector "input[value='#{value}']"
+      end
+    end
+
+    context 'with value from object' do
+      before { resource.email = email }
```
Comment (moshinaan) [inline spec/components/tailwinds/form/builder_spec.rb:29]: i think its better to add empty line after line 29

File: spec/components/tailwinds/form/builder_spec.rb [hunk 3]
```diff
@@ -7,13 +7,32 @@
   let(:builder) { Tailwinds::Form::Builder.new :user, resource, view, {} }
 
   describe '#text_field' do
-    let(:output) do
-      builder.text_field :email
+    context 'default' do
+      let(:output) { builder.text_field :email }
+
+      it do
```
Comment (moshinaan) [inline spec/components/tailwinds/form/builder_spec.rb:13]:
```suggestion
      it doit 'gets default value' do
```

File: spec/components/tailwinds/form/builder_spec.rb [hunk 4]
```diff
@@ -51,15 +70,33 @@
   end
 
   describe '#select' do
-    let(:output) do
-      builder.select :role, %i[admin user]
+    context 'default' do
+      let(:output) { builder.select :role, %i[admin user] }
+
+      it do
```
Comment (moshinaan) [inline spec/components/tailwinds/form/builder_spec.rb:76]: maybe its better to add the name for checks as in `spec/components/tailwinds/form/builder_spec.rb`

File: spec/dummy/app/models/user.rb [hunk 5]
```diff
@@ -4,5 +4,7 @@
 class User < ApplicationRecord
   has_many :posts
 
-  attr_reader :password, :file, :role
+  attr_reader :password, :file
+  # :reek:Attribute { enabled: false }
+  attr_accessor :role
```
Comment (moshinaan) [inline spec/dummy/app/models/user.rb:9]: why did you change  `attr_reader` to `attr_reader` here?

File: app/components/tailwinds/form/builder.rb (hunk 1 above)
Comment (kalashnikovisme) [reply app/components/tailwinds/form/builder.rb:45]: The `input` method looks strange for usual Ruby application cuz it's some kind of functional programming. So, I decided to make more readable and easy-to-understand :slightly_smiling_face:

File: spec/components/tailwinds/form/builder_spec.rb (hunk 2 above)
Comment (kalashnikovisme) [reply spec/components/tailwinds/form/builder_spec.rb:29]: done!

File: spec/components/tailwinds/form/builder_spec.rb (hunk 3 above)
Comment (kalashnikovisme) [reply spec/components/tailwinds/form/builder_spec.rb:13]: Changed :slightly_smiling_face:

I have not accepted your suggestion because it contains typos :slightly_smiling_face:

File: spec/components/tailwinds/form/builder_spec.rb (hunk 4 above)
Comment (kalashnikovisme) [reply spec/components/tailwinds/form/builder_spec.rb:77]: done

File: spec/dummy/app/models/user.rb (hunk 5 above)
Comment (kalashnikovisme) [reply spec/dummy/app/models/user.rb:9]: We need it for the last test in `spec/components/tailwinds/forms/builder_spec.rb`. Line 95

**Review by moshinaan — APPROVED** [review]
